### ISSUE TITLE
Move params to context of Router

### ISSUE THREAD

**Issue by NaNCat** [ItalyPaleAle/svelte-spa-router#16]
This will allow you to get the params in any nested component, not just a child component.
What do you think?

**Comment by ItalyPaleAle** [comment]
This is something I had explored (very briefly) before. My feeling was that to keep things simple (my philosophy with this Router) I could just let developers implement the context in their app. My understanding is that contexts are app-wide states and so aren’t passed just to children, but also to the parent, siblings...

I am open to re-consider this however. What do you see as the main benefit for implementing this in the router itself? (Possibly as a non-breaking change)

Another option could just be write a paragraph in the “advanced docs” mentioning contexts.

**Comment by NaNCat** [comment]
I'm sorry, I changed the structure of my project a little bit and was able to solve my problem.

For clarity:
I have a component that should create a link depending on the parameters passed into it. To do this, I received params from Router and then passed them to the component via all parents. Thinking about how to get rid of the problem, I changed the routing so that I could use the stored value in the $location.
